web_external/User/InviteUserView.js

```diff
@@ -2,28 +2,40 @@ isic.views.InviteUserView = isic.View.extend({
     events: {
         'submit #isic-user-invite-form': function (event) {
             event.preventDefault();
-            // this.$('#isic-user-invite-submit').prop('disabled', true);
+            this.$('#isic-user-invite-submit').prop('disabled', true);
+
+            var data = {
+                login: this.$('#isic-user-invite-new-login').val(),
+                email: this.$('#isic-user-invite-new-email').val(),
+                firstName: this.$('#isic-user-invite-new-firstname').val(),
+                lastName: this.$('#isic-user-invite-new-lastname').val()
+            };
+
+            var validityPeriod = this.$('#isic-user-invite-period').val();
+            if ($.trim(validityPeriod).length > 0) {
+                data.validityPeriod = validityPeriod;
+            }
 
             girder.restRequest({
                 path: 'user/invite',
-                data: {
-                    login: this.$('#isic-user-invite-new-login').val(),
-                    email: this.$('#isic-user-invite-new-email').val(),
-                    firstName: this.$('#isic-user-invite-new-firstname').val(),
-                    lastName: this.$('#isic-user-invite-new-lastname').val()
-                },
+                data: data,
                 type: 'POST',
                 error: null
             })
-                .done(function (resp) {
-                    console.log('success', resp);
-                })
-                .fail(function (resp) {
-                    isic.showAlertDialog({
-                        text: '<h4>Error sending invite</h4><br>' + _.escape(resp.responseJSON.message),
-                        escapedHtml: true
-                    });
+            .done(function (resp) {
+                resp.newUser.email = data.email;
+                girder.events.trigger('g:navigateTo', isic.views.InvitationConfirmationView, {
```

Comment:
Now I'm not so sure this is the best way. The code separate is nice, but this breaks the "Invite New User" menu when on the "Invitation sent" view. Because of that I think one of the other options might be preferable.